### PR TITLE
added collector dummy

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2004.

### CODE REVIEW

File: collector.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+type Collector struct {
+	restClient *resty.Client
+
+	bundles []Bundle
+
+	endpoints []url.URL
+}
+
+func NewCollector(config CollectorConfig) (*Collector, error) {
```
Comment (xh3b4sd) [inline collector.go]: We have to collect version bundles from all the microservices and operators that expose any. Therefore we need to request all version endpoints that are of interest for us. The idea here is to have a collector to do this job.

File: collector.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+		endpoints: config.Endpoints,
+	}
+
+	return c, nil
+}
+
+func (c *Collector) Bundles() []Bundle {
+	return c.bundles
+}
```
Comment (xh3b4sd) [inline collector.go]: The idea is to have a dumb getter to return the cached bundles found so far. The flat list of bundles contains any bundles being exposed by any of the configured endpoints. The result provided by the collector can then be given to `versionbundle.Aggregate`. The result of this in turn can be used to compute distribution types we introduced in a former PR.

File: collector.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+// TODO
+func (c *Collector) Collect() error {
+	return nil
+}
```
Comment (xh3b4sd) [inline collector.go]: The idea is to periodically collect version bundles from the configured endpoints. The collector is configured with a common rest client implementation which in turn can be configured with a fake transport to be able to do some unit testing.

File: collector.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		RestClient: nil,
+
+		Endpoints: nil,
+	}
+}
+
+type Collector struct {
+	restClient *resty.Client
+
+	bundles []Bundle
```
Comment (JosephSalisbury) [inline collector.go]: I don't get the ordering here - should `bundles` be at the bottom? If `endpoints` is configuration, and `bundles` is internal.

**Comment by JosephSalisbury** [comment]
> The idea is to periodically collect version bundles from the configured endpoints. 

This needs clarification for me. The service calls `Collect` periodically?

File: collector.go (hunk 4 above)
Comment (xh3b4sd) [reply collector.go]: - dependencies
- internals
- settings

**Comment by xh3b4sd** [comment]
> he service calls Collect periodically?

I thought we aggregate within our API microservice. So the API periodically calls `Collect` to keep the bundles cache in sync and on each request `Bundles` can just be used without network delays.

File: collector.go (hunk 4 above)
Comment (JosephSalisbury) [reply collector.go]: Ah, I normally roll with internals and settings the other way around. all good.

**Comment by JosephSalisbury** [comment]
Yeah cool, this makes sense to me.

**Pull request merged**